tools/payloads/configbytes/apk_configbytes_extract.rb

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env ruby
+# -*- coding: binary -*-
+
+class ExtractError < StandardError; end
+class HelpError < StandardError; end
+class UsageError < ExtractError; end
+
+require 'rex'
+require 'tmpdir'
+require 'optparse'
+require 'open3'
+require 'nokogiri'
+require 'fileutils'
+
+def run_cmd(bin, args, verbose)
+  begin
+    path = Rex::FileUtils.find_full_path(bin)
+    cmd = [path, args].join(" ")
+    stdin, stdout, stderr = Open3.popen3(cmd)
+
+    if verbose
+      $stdout.puts "[+] Running: #{cmd}"
+      $stdout.puts stdout.read + stderr.read
+    end
+
+    return stdout.read + stderr.read
+  rescue Errno::ENOENT
+    return nil
+  end
+end
+
+def parse_manifest(manifest_file)
+  File.open(manifest_file, "rb"){|file|
+    data = File.read(file)
+    return Nokogiri::XML(data)
+  }
+end
+
+def extract_data(java_file, text_file)
+  file_data = File.read(java_file)
+  mid = file_data.split("[]{")
+  bytes = mid[1].split("}")
+  bytes[0]
+
+  File.open(text_file, 'wb') do |f|
+    f.write(bytes[0])
+  end
+
+  $stdout.puts "[+] Saved as: #{text_file}"
+end
+
+def check_tools(verbose)
+  # CHECK IF ALL TOOLS ARE INCLUDED BEFORE STARTING
+  tools = ["unzip", "d2j-dex2jar", "java", "apktool"]
+
+  tools.each do |tool|
+    path = Rex::FileUtils.find_full_path(tool)
+    if path && ::File.file?(path)
+      $stdout.puts "[+] Tool present: #{path}" if verbose
+    else
+      raise RuntimeError, "[-] #{tool} command not found."
+    end
+  end
+end
+
+def parse_args(args)
+  opts = {}
+  opt = OptionParser.new
+  banner = "apk_configbytes_extract - a tool to extract configbytes from an android payload.\n"
+  banner << "Usage: #{$0} < -a apk-file -o path -j path > [options]\n"
+  banner << "Example: #{$0} -a metasploit.apk -o /root/configbytes.txt -j /root/fernflower.jar"
+  opt.banner = banner
+  opt.separator('')
+  opt.separator('Options:')
+
+  opt.on('-a', '--apk             <path>', String, 'Specify apk to extract configbytes') do |a|
+    opts[:apk] = a
+  end
+
+  opt.on('-o', '--out             <path>', String, 'Save configbytes to a file') do |o|
+    opts[:out] = o
+  end
+
+  opt.on('-j', '--jar             <path>', String, 'Specify fernflower path') do |j|
+    opts[:jar] = j
+  end
+
+  opt.on('-v', '--verbose', 'Displays verbose output') do
+    opts[:verbose] = true
+  end
+
+  opt.on('-k', '--keep', 'Keep working directory') do
+    opts[:keep] = true
+  end
```

Comment:
I'm not sure this is necessary. When the Ruby script exits the working directory that the user had within their shell environment.

Reply:
This option is to keep the directory where all work is being done. In case further analysis of the decompiled files, such as viewing java files, is required.

Reply:
If it's referring the the analysis directory where the files are extracted, it should say that instead of the "working" directory which is it's own thing in the context of a CLI tool.